### ISSUE TITLE
No profiles were found iff automatic signing is turned on

### ISSUE THREAD

**Issue by kdong007** [fastlane/fastlane#10538]
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                           |
| --------------------------- | ----------------------------------------------- |
| OS                          | 10.12.6                                         |
| Ruby                        | 2.2.4                                           |
| Bundler?                    | false                                           |
| Git                         | git version 2.14.1                              |
| Installation Source         | /usr/local/lib/fastlane_lib/bundle/bin/fastlane |
| Host                        | Mac OS X 10.12.6 (16G29)                        |
| Ruby Lib Dir                | /usr/local/lib/fastlane_lib/bundle/lib          |
| OpenSSL Version             | OpenSSL 1.0.2g  1 Mar 2016                      |
| Is contained                | false                                           |
| Is homebrew                 | false                                           |
| Is installed via Fabric.app | true                                            |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/     |
| Xcode Version               | 9.0                                             |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
fastlane_version "2.60.1"

default_platform :ios

platform :ios do
  #
  # Learn more here: https://docs.fastlane.tools/getting-started/ios/beta-deployment/
  #             and: https://docs.fastlane.tools/getting-started/android/beta-deployment/
  #
  lane :beta do |values|
    # Fabric generated this lane for deployment to Crashlytics Beta
    # set 'export_method' to 'ad-hoc' if your Crashlytics Beta distribution uses ad-hoc provisioning
    gym(
    	scheme: 'ReactNativeApp', 
    	export_options: {
    		method: 'ad-hoc',
    		provisioningProfiles: {
				"com.xx.xxx" => "xxx Ad Hoc"
			}
		},
    )

    emails = values[:test_email] ? values[:test_email] : 
    	[
    	]
    groups = values[:test_email] ? nil : nil # You can define groups on the web and reference them here

    crashlytics(api_token: 'xxxx',
             build_secret: 'xxxx',
                   emails: emails,
                   groups: groups,
                    notes: 'Distributed with fastlane', # Check out the changelog_from_git_commits action
            notifications: true) # Should this distribution notify your testers via email?

    # for all available options run `fastlane action crashlytics`

    # You can notify your team in chat that a beta build has been uploaded
    # slack(
    #   slack_url: "https://hooks.slack.com/services/YOUR/TEAM/INFO"
    #   channel: "beta-releases",
    #   message: "Successfully uploaded a beta release - see it at https://fabric.io/_/beta"
    # )
  end
end

```
</details>

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.60.1  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.5        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.2        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| terminal-notifier         | 1.8.0        |
| unicode-display_width     | 1.3.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.3.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.2        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.5        |
| excon                     | 0.59.0       |
| faraday                   | 0.13.1       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| json                      | 1.8.1        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.12.2       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.2.1        |
| bundler                   | 1.15.4       |
| faraday_middleware        | 0.12.2       |
| uber                      | 0.1.0        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.1        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| jwt                       | 1.5.6        |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.3        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.13.6       |
</details>

*generated on:* **2017-10-08**
</details>

If I turn OFF automatic signing in xcode and manually select my Ad Hoc profile, `fastlane beta` works.

If I turn ON automatic signing, `fastlane beta` breaks. It can't find my profile anymore. 
```Error Domain=IDEProfileLocatorErrorDomain Code=1 "No profiles for 'com.xx.xxx' were found" UserInfo={NSLocalizedDescription=No profiles for 'com.xx.xxx' were found, NSLocalizedRecoverySuggestion=Xcode couldn't find any iOS Ad Hoc provisioning profiles matching 'com.xx.xxx'. Automatic signing is disabled and unable to generate a profile. To enable automatic signing, pass -allowProvisioningUpdates to xcodebuild.```
What's more interesting is the error log even says `Automatic signing is disabled` LOL?

**Comment by kdong007** [comment]
adding this fixed the issue 

gym(
    ....
     export_xcargs: "-allowProvisioningUpdates"
)

although i have no idea of what that is doing... LOL GG

**Comment by tomerpeled** [comment]
I'm also encountering this issue. My automatic build process used to work perfect and in the last day it failed after I had to approve some xcode new user agreement.
Now my build process is failing with the mentioned error.
Anything that can be done?

**Comment by kdong007** [comment]
@tomerpeled  have you tried adding `export_xcargs: "-allowProvisioningUpdates"`

**Comment by tomerpeled** [comment]
Found the issue:
I ran the "fastlane beta" command on the terminal and found out that during the provision profile phase it had to request access to the provisioning profile. So it showed up a pop-up requesting the permissions to access to the provisioning profile on that mac.

Of course that when I ran this process via our CI, the pop-up cannot be approved so it failed our CI builds....

For summary, if Apple is changing it's user agreement, you'll probably have to re-grant permissions do several procedure which already got permissions in the past.
